File: BackEnd_ExpressJS/app/models/userTypeHasItemMenu.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const ItemMenu = require('./itemMenu');
const UserType = require('./userType');

const UserTypeHasItemMenu = sequelize.define('tblUserTypeHasItemMenu', {
        ideUserTypeHasItemMenu: {
            type: DataTypes.STRING,
            primaryKey: true,
        },
        ideUserType: {
            type: DataTypes.BIGINT,
            allowNull: false,
            references: {
              model: UserType,
              key: 'ideUserType',
            },
            onUpdate: 'CASCADE',
            onDelete: 'CASCADE',
        },
        ideItemMenu: {
            type: DataTypes.BIGINT,
            allowNull: false,
            references: {
              model: ItemMenu,
              key: 'ideItemMenu',
            },
            onUpdate: 'CASCADE',
            onDelete: 'CASCADE',
        },
        timeStamp: {
            type: DataTypes.DATE,
            allowNull: false,
        },
    }, {
        // Activar/Desactivar las marcas de tiempo
        timestamps: false,
        // Usar el nombre de la tabla tal como está en la base de datos
        freezeTableName: true,
    });
    
UserTypeHasItemMenu.belongsTo(ItemMenu, { foreignKey: 'ideItemMenu' });

module.exports = UserTypeHasItemMenu;

File: BackEnd_Django/api/myModels/userTypeHasItemMenu.py
from django.db import models
from .userType import UserType
from .itemMenu import ItemMenu

class UserTypeHasItemMenu(models.Model):
    ideUserTypeHasItemMenu = models.AutoField(primary_key=True)
    ideUserType = models.ForeignKey(UserType, on_delete=models.CASCADE)
    ideItemMenu = models.ForeignKey(ItemMenu, on_delete=models.CASCADE)
    timeStamp = models.DateTimeField(auto_now_add=True)

    class Meta:
        db_table = 'tblUserTypeHasItemMenu'
        # managed = False  # Para desactivar la creación de migraciones y la administración de la tabla

    def __str__(self):
        return f'{self.ideUserType} - {self.ideItemMenu}'
